alchemiscale_fah/tests/integration/compute/test_compute_client.py

```diff
@@ -0,0 +1,272 @@
+
+        fah_project_ = FahProject.parse_raw(
+            client.get_project_file_to_bytes(project_id, project_file).decode("utf-8")
+        )
+
+        assert fah_project_ == fah_project
+
+    def test_create_run_file_from_bytes(self, fah_adaptive_sampling_client):
+        client: FahAdaptiveSamplingClient = fah_adaptive_sampling_client
+
+        project_id = 90001
+        run_id = 0
+
+        project_data = ProjectData(
+            core_id=0x23,
+            contact="lol@no.int",
+            atoms=10000,
+            credit=5000,
+        )
+
+        client.create_project(project_id, project_data)
+
+        # create test file
+        content = b"brown cow how now"
+        dest = "testfile.out"
+
+        # try creating file
+        client.create_run_file_from_bytes(project_id, run_id, content, dest)
+
+        # try getting it back
+        content_ = client.get_run_file_to_bytes(project_id, run_id, dest)
+
+        assert content_ == content
+
+    def test_create_clone(self, fah_adaptive_sampling_client):
+        client: FahAdaptiveSamplingClient = fah_adaptive_sampling_client
+
+        project_id = 90001
+        run_id = 0
+        clone_id = 0
+
+        project_data = ProjectData(
+            core_id=0x23,
+            contact="lol@no.int",
+            atoms=10000,
+            credit=5000,
+        )
+
+        client.create_project(project_id, project_data)
+
+        client.create_clone(project_id, run_id, clone_id)
+
+        jobdata = client.get_clone(project_id, run_id, clone_id)
+
+        assert jobdata.project == project_id
+        assert jobdata.run == run_id
+        assert jobdata.clone == clone_id
+
+        assert jobdata.state == "READY"
+        assert jobdata.core == 0x23
+
+    def test_get_clone(self, fah_adaptive_sampling_client):
+        client: FahAdaptiveSamplingClient = fah_adaptive_sampling_client
+
+        project_id = 90001
+        run_id = 0
+        clone_id = 0
+
+        project_data = ProjectData(
+            core_id=0x23,
+            contact="lol@no.int",
+            atoms=10000,
+            credit=5000,
+        )
+
+        client.create_project(project_id, project_data)
+
+        client.create_clone(project_id, run_id, clone_id)
+
+        jobdata = client.get_clone(project_id, run_id, clone_id)
+
+        assert jobdata.project == project_id
+        assert jobdata.run == run_id
+        assert jobdata.clone == clone_id
+
+        assert jobdata.state == "READY"
+        assert jobdata.core == 0x23
+
+        # now, let the job "finish"
+        client._finish_clone_mock_ws(project_id, run_id, clone_id)
+
+        jobdata = client.get_clone(project_id, run_id, clone_id)
+
+        assert jobdata.state == "FINISHED"
+
+    def test_create_clone_file(self, fah_adaptive_sampling_client):
+        client: FahAdaptiveSamplingClient = fah_adaptive_sampling_client
+
+        project_id = 90001
+        run_id = 0
+        clone_id = 0
+
+        project_data = ProjectData(
```

Comment:
project creation is done many times, better to put in a fixture if possible

Reply:
Moved most of these to fixture.